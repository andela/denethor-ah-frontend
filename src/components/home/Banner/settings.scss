// spacing


@mixin overlay-background($overlay, $image: false) {
  @if ($image) {
    background-image: linear-gradient(0deg, rgba(0,0,0,$overlay), rgba(0,0,0,$overlay)), url($image);
  } @else {
    background-image: linear-gradient(0deg, rgba(0,0,0,$overlay), rgba(0,0,0,$overlay));
  }
}

@mixin reset() {
  margin: 0;
  padding: 0;
}

button {
  cursor: pointer;
}

button:disabled {
  cursor: default;
}