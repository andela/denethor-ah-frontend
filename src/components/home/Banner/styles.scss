@import './settings';
@import '../../../styles/base/settings';

.banner {
  @include overlay-background(0.5, '/assets/images/banner.png');
  background-size: cover;
  background-repeat: no-repeat;
  display: grid;
  grid-template-columns: 3fr 5fr;
  padding: 6.4rem 1.92rem;
}

.banner__yellow-text {
  color: darken(#E3B154, 10%);
}

.banner__off-white-text {
  color: darken(white, 25%);
}

.banner__slogan {
  align-items: center;
  display: flex;
  justify-content: center;
  text-shadow: 0.1rem 0.1rem 0.1rem #000;

  > div h1:first-of-type {
    color: #EBB03D;
    font-family: 'Montserrat-Bold';
    margin: 0;
  }

  > div h1:last-of-type {
    font-family: 'Century-Gothic-Bold';
  }
  
  > div {
    padding: 3.2rem 0;
  }

  button {
    background: linear-gradient(135deg, #D56E49, darken(#D56E49, 10%));
    border: none;
    border-radius: 2rem;
    box-shadow: -0.1rem -0.1rem 0.1rem darken(#D56E49, 30%) inset, 0.1rem 0.1rem 0.1rem lighten(#D56E49, 10%) inset;
    font-size: 1.28rem;
    font-family: 'Montserrat-SemiBold';
    height: 2.56rem;
    margin: 0.64rem;
    width: 9.6rem;
    color: white;
  }

  button:hover {
    box-shadow:
      -0.1rem -0.1rem 0.1rem darken(#D56E49, 50%) inset,
      -0.2rem -0.2rem 0.2rem darken(#D56E49, 30%) inset,
      0.2rem 0.2rem 0.2rem darken(white, 70%),
      0.2rem 0.2rem 0.2rem lighten(#D56E49, 10%) inset,
      0.1rem 0.1rem 0.1rem lighten(#D56E49, 20%) inset;
    color: darken(white, 10%);
  }

  h1 {
    font-size: 2.56rem;
    color: white;
  }
}

.py-2 {
  padding: $m-size 0;
}


.banner__screen {
  @include overlay-background(0.5);
  border-radius: 0.5rem;
  overflow-x: hidden;
  position: relative;
  .flex {
    position: absolute;
    width: 100%;
  }
}

@media (max-width: 60rem) {
  .banner {
    grid-template-columns: 5fr 5fr;
    padding: 1.6rem;
  }

  .banner__slogan {
    > div h1 {
      font-size: 2rem;
      padding-right: 0.8rem;
    }
  }
}

@media (max-width: 40rem) {
  .banner {
    background: inherit;
    display: block;
    height: auto;
    padding: 0;
  }

  .banner__slogan {
    @include overlay-background(0.5, '/assets/images/banner-mobile.jpg');
    background-size: cover;
    background-repeat: no-repeat;
    text-align: center;
    h1 {
      font-size: 2.4rem;
      border-bottom: 0.1rem solid white;
      margin-bottom: 3.2rem;
      margin-top: 6.4rem;
      line-height: 1.2;
      padding-bottom: 2.4rem;
    }

    > div h1 {
      font-size: 2.4rem;
    }

    button {
      margin-top: 3.2rem;
    }
  }

  .banner__screen {
    background: none;
  }

  .hidden-on-mobile {
    display: none;
  }

  .banner__yellow-text-mobile {
    color: darken(#E3B154, 10%);
  }

  .banner__off-white-text {
    color: darken(white, 75%);
  }
}

.background-enter {
  transform: translate(100%);
  opacity: 0.8;
}
.background-enter.background-enter-active {
  transform: translate(0%);
  transition: transform 300ms cubic-bezier(.6,.67,.5,.25);
}
.background-leave {
  transform: translate(0%);
}
.background-leave.background-leave-active {
  opacity: 0.5;
  transform: translate(-100%);
  transition: transform 200ms cubic-bezier(.6,.67,.5,.25);
}