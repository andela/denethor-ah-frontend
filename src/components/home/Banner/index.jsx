import React from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import Stats from './Stats/Stats';
import Signup from './Signup/Signup';
import Login from './Login/Login';
import './styles.scss';

const sides = {
  Stats, Signup, Login
};

const Banner = ({ side, handleSignup }) => {
  const Side = sides[side];
  return (
    <div className="banner">
      <div className="banner__slogan">
        <div>
          <h1 className="banner__yellow-text hidden-on-mobile">The place <br /> writers love...</h1>
          <h1><span className="banner__yellow-text-mobile">Great writers,</span><br />Quality Content</h1>
          <button onClick={handleSignup}>Get Started</button>
        </div>
      </div>
      <div className="banner__screen">
        <ReactCSSTransitionGroup
          transitionName="background"
          transitionEnterTimeout={500}
          transitionLeaveTimeout={400}
        >
          <Side key={side}/>
        </ReactCSSTransitionGroup>
      </div>
    </div>
  );
};

export default Banner;
