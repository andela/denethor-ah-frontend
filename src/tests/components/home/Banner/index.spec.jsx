import React from 'react';
import { mount } from 'enzyme';
import Banner from '../../../../components/home/Banner';

test('Should render banner with stats correctly', () => {
  const wrapper = mount(<Banner side={'Stats'}/>);
  expect(wrapper).toMatchSnapshot();
});

test('Should render banner with login correctly', () => {
  const wrapper = mount(<Banner side={'Login'}/>);
  expect(wrapper).toMatchSnapshot();
});

test('Should render banner with signup correctly', () => {
  const wrapper = mount(<Banner side={'Signup'}/>);
  expect(wrapper).toMatchSnapshot();
});
