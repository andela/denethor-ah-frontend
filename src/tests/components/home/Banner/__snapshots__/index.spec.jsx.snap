// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render banner with login correctly 1`] = `
<div
  className="banner"
>
  <div
    className="banner__slogan"
  >
    <div>
      <h1
        className="banner__yellow-text hidden-on-mobile"
      >
        The place 
        <br />
         writers love...
      </h1>
      <h1>
        <span
          className="banner__yellow-text-mobile"
        >
          Great writers,
        </span>
        <br />
        Quality Content
      </h1>
      <button
        onClick={[Function]}
      >
        Get Started
      </button>
    </div>
  </div>
  <div
    className="banner__screen banner__screen--login"
  >
    <CSSTransitionGroup
      transitionAppear={true}
      transitionAppearTimeout={800}
      transitionEnter={true}
      transitionEnterTimeout={800}
      transitionLeave={true}
      transitionLeaveTimeout={400}
      transitionName="background"
    >
      <Connect(Login)
        key="Login"
      />
    </CSSTransitionGroup>
  </div>
</div>
`;

exports[`Should render banner with signup correctly 1`] = `
<div
  className="banner"
>
  <div
    className="banner__slogan"
  >
    <div>
      <h1
        className="banner__yellow-text hidden-on-mobile"
      >
        The place 
        <br />
         writers love...
      </h1>
      <h1>
        <span
          className="banner__yellow-text-mobile"
        >
          Great writers,
        </span>
        <br />
        Quality Content
      </h1>
      <button
        onClick={[Function]}
      >
        Get Started
      </button>
    </div>
  </div>
  <div
    className="banner__screen banner__screen--signup"
  >
    <CSSTransitionGroup
      transitionAppear={true}
      transitionAppearTimeout={800}
      transitionEnter={true}
      transitionEnterTimeout={800}
      transitionLeave={true}
      transitionLeaveTimeout={400}
      transitionName="background"
    >
      <_default
        key="Signup"
      />
    </CSSTransitionGroup>
  </div>
</div>
`;

exports[`Should render banner with stats correctly 1`] = `
<div
  className="banner"
>
  <div
    className="banner__slogan"
  >
    <div>
      <h1
        className="banner__yellow-text hidden-on-mobile"
      >
        The place 
        <br />
         writers love...
      </h1>
      <h1>
        <span
          className="banner__yellow-text-mobile"
        >
          Great writers,
        </span>
        <br />
        Quality Content
      </h1>
      <button
        onClick={[Function]}
      >
        Get Started
      </button>
    </div>
  </div>
  <div
    className="banner__screen banner__screen--stats"
  >
    <CSSTransitionGroup
      transitionAppear={true}
      transitionAppearTimeout={800}
      transitionEnter={true}
      transitionEnterTimeout={800}
      transitionLeave={true}
      transitionLeaveTimeout={400}
      transitionName="background"
    >
      <Connect(Stats)
        key="Stats"
      />
    </CSSTransitionGroup>
  </div>
</div>
`;
